src/main/java/bio/terra/cloudfiletodatastore/testrunner/cloudfunctions/TestRunnerStreamingApp.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.cloudfiletodatastore.testrunner.cloudfunctions;
+
+import bio.terra.cloudfiletodatastore.FileUploadedMessage;
+import bio.terra.cloudfunctions.common.App;
+import bio.terra.cloudfunctions.utils.BigQueryUtils;
+import bio.terra.cloudfunctions.utils.GcsUtils;
+import bio.terra.cloudfunctions.utils.MediaTypeUtils;
+import com.google.common.io.Files;
+import java.io.InputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.compressors.CompressorInputStream;
+
+public class TestRunnerStreamingApp extends App {
+  private static final Logger logger = Logger.getLogger(TestRunnerStreamingApp.class.getName());
+
+  public TestRunnerStreamingApp(FileUploadedMessage fileUploadedMessage) {
+    super(fileUploadedMessage);
+  }
+
+  @Override
+  public void process() throws Exception {
```

Comment:
It looks like we're not closing `archiveInputStream`.  Also similar to my comment in `TestRunnerStreamingFunction` you can convert from an if else to if.. return and reduce indentation and increase readability if you flip the logic while reading the archiveInputStream.

```
    while ((archiveEntry = archiveInputStream.getNextEntry()) != null) {
      if (archiveEntry.isDirectory()) {
        continue;
      }

      if (!Files.getNameWithoutExtension(archiveEntry.getName()).equals(table)) {
        logger.log(
                Level.WARNING,
                String.format(
                        "Processing skipped for project %s: bucket %s and resource name %s.",
                        projectId, sourceBucket, resourceName));
        continue;

      }
      logger.log(
              Level.INFO,
              String.format(
                      "Processing %s (%s bytes) for streaming to BQ table %s",
                      archiveEntry.getName(),
                      archiveEntry.getSize(),
                      Files.getNameWithoutExtension(archiveEntry.getName())));

      byte[] data = archiveInputStream.readAllBytes();
      BigQueryUtils.streamToBQ(projectId, dataSet, table, data);
    }

```

Reply:
Reorganized the code for readability. Wrap code in try-catch to close stream automatically.